pdata/plog/json.go

```diff
@@ -0,0 +1,66 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plog // import "go.opentelemetry.io/collector/pdata/plog"
+
+import (
+	"bytes"
+
+	"github.com/gogo/protobuf/jsonpb"
+
+	"go.opentelemetry.io/collector/pdata/internal"
+	otlplogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1"
+	"go.opentelemetry.io/collector/pdata/internal/otlp"
+)
+
+// NewJSONLogsMarshaler returns a LogsMarshaler. Marshals to OTLP json bytes.
+func NewJSONLogsMarshaler() LogsMarshaler {
```

Comment:
I would consider to remove the "Logs" from the `NewJSONLogsMarshaler`? What do you think? Same for proto.

Reply:
Makes sense. In that case we also rename interfaces `plog.LogsMarshaler` and `plog.LogsUnmarshaler` to `plog.Marshaler` and `plog.Unmarshaler`, but keep names of all the methods like `MarshalLogs` and `UnmarshalLogs`. Same goes to `plog.Sizer` as pointed in the discussion above.

How does it sound?

Reply:
I like it. In this PR was referring to the `NewJSONLogsMarshaler` -> `NewJSONMarshaler`

Reply:
But yes, you mentioned all the changes I was thinking about in your comments :)

Reply:
Applied this one and other suggestions.